packages/picolisp/build.sh

```diff
@@ -1,84 +1,79 @@
 TERMUX_PKG_HOMEPAGE=https://picolisp.com/wiki/?home
 TERMUX_PKG_DESCRIPTION="Lisp interpreter and application server framework"
 TERMUX_PKG_LICENSE="MIT"
-# TERMUX_PKG_SRCDIR is overriden below
 TERMUX_PKG_LICENSE_FILE="../COPYING"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION=20.7.24
-TERMUX_PKG_REVISION=2
-# We use our source copies since old version snapshots are not kept on main site.
-TERMUX_PKG_SRCURL=https://github.com/termux/distfiles/releases/download/2021.01.04/picolisp_${TERMUX_PKG_VERSION}.tar.gz
-TERMUX_PKG_SHA256=72e8d394ad32a6659210ac2cc4b7bb736dc9b94c6ac8d6296506b6dfdc92f90c
-TERMUX_PKG_DEPENDS="libcrypt, openssl"
+TERMUX_PKG_VERSION=21.12
+TERMUX_PKG_SRCURL=http://archive.ubuntu.com/ubuntu/pool/universe/p/picolisp/picolisp_$TERMUX_PKG_VERSION.orig.tar.gz
+TERMUX_PKG_SHA256=a06838236b7f5b52c5d587d32d31627f73cdb9775cc02a80f2cdaedd12888c7d
+# Clang is required for compiling sysdefs.c and running the binary which is not possible while cross compiling
+TERMUX_PKG_DEPENDS="clang, libcrypt, libffi, openssl, readline"
 TERMUX_PKG_BUILD_IN_SRC=true
-# arm and i686: The c code uses gcc-specific "variable length array in structure":
-# x86_64: The assembly is not position-independent:
-TERMUX_PKG_BLACKLISTED_ARCHES="arm, i686, x86_64"
 
-termux_step_pre_configure() {
-	# Certain packages are not safe to build on device because their
-	# build.sh script deletes specific files in $TERMUX_PREFIX.
-	if $TERMUX_ON_DEVICE_BUILD; then
-		termux_error_exit "Package '$TERMUX_PKG_NAME' is not safe for on-device builds."
-	fi
-
-	# Validate that we have the right version:
-	grep -q "Version ${TERMUX_PKG_VERSION//./ }" src64/version.l || {
-		termux_error_exit "Picolisp version needs to be bumped"
-	}
-
-	if [ $TERMUX_ARCH_BITS = 64 ]; then
-		cd $TERMUX_PKG_SRCDIR
-		if [ $TERMUX_ARCH = "aarch64" ]; then
-			export TERMUX_PKG_EXTRA_MAKE_ARGS=arm64.android
-		else
-			termux_error_exit "Unsupported arch: $TERMUX_ARCH"
-		fi
-		TERMUX_PKG_SRCDIR=$TERMUX_PKG_SRCDIR/src64
-	else
-		TERMUX_PKG_SRCDIR=$TERMUX_PKG_SRCDIR/src
-	fi
-	TERMUX_PKG_BUILDDIR=$TERMUX_PKG_SRCDIR
-	ORIG_CFLAGS="$CFLAGS"
-	CFLAGS+=" -c $LDFLAGS $CPPFLAGS"
+termux_step_post_get_source() {
+	TERMUX_PKG_SRCDIR+="/src"
 }
 
-termux_step_make_install() {
-	cd $TERMUX_PKG_SRCDIR/
+termux_step_make() {
+	cd $TERMUX_PKG_SRCDIR
+	opt -O3 --mtriple=$TERMUX_HOST_PLATFORM -o base.bc base.ll
```

Comment:
The `opt` binary isn't shipped by NDK. Seems like we'll need to install the mighty llvm on host